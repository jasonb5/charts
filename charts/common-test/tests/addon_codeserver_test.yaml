suite: test codeserver addon
values:
  - values/common.yaml
templates:
  - app.yaml
chart:
  appVersion: 0.1.0
tests:
  - it: should overwrite default
    set:
      workload: deployment

      addons:
        codeserver:
          enabled: true

          image:
            tag: 4.16.1
            pullPolicy: Always

          ingress:
            enabled: true
            hosts:
              - name: codeserver
                host: code.domain.io
                path: /code
                pathType: Prefix

          persistence:
            config:
              enabled: true
              type: pvc
              size: 20G
            data:
              enabled: true
              type: hostpath
              path: /data
    asserts:
      - matchSnapshot:
          path: spec
        documentIndex: 3
      - matchSnapshot:
          path: spec
        documentIndex: 2
      - matchSnapshot:
          path: spec.ports
        documentIndex: 1
      - matchSnapshot:
          path: spec.template.spec.containers[1]
        documentIndex: 0
      - hasDocuments:
          count: 4
  - it: should pass
    set:
      workload: deployment

      addons:
        codeserver:
          enabled: true
    asserts:
      - matchSnapshot:
          path: spec.ports
        documentIndex: 1
      - matchSnapshot:
          path: spec.template.spec.containers[1]
        documentIndex: 0
      - hasDocuments:
          count: 2
