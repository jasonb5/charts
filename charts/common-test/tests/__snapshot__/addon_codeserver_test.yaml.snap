should overwrite default:
  1: |
    accessModes:
      - ReadWriteOnce
    resources:
      requests:
        storage: 20G
    volumeMode: Filesystem
  2: |
    rules:
      - host: code.domain.io
        http:
          paths:
            - backend:
                service:
                  name: RELEASE-NAME-common-test
                  port:
                    number: 8443
              path: /
              pathType: ImplementationSpecific
  3: |
    - name: codeserver
      port: 8443
      protocol: TCP
      targetPort: 8443
  4: |
    image: ghcr.io/linuxserver/code-server:4.16.1
    imagePullPolicy: Always
    name: codeserver
    ports:
      - containerPort: 8443
        name: codeserver
        protocol: TCP
    volumeMounts:
      - mountPath: /config
        name: config
      - mountPath: /data
        name: data
should pass:
  1: |
    - name: codeserver
      port: 8443
      protocol: TCP
      targetPort: 8443
  2: |
    image: ghcr.io/linuxserver/code-server:4.17.1
    name: codeserver
    ports:
      - containerPort: 8443
        name: codeserver
        protocol: TCP
    volumeMounts: null
